{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "machine_shape": "hm",
      "gpuType": "V100"
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    },
    "accelerator": "GPU"
  },
  "cells": [
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "GpEZSwNB6RP5",
        "outputId": "42754c78-f233-4a5e-ae34-c7f46f35bcc0"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  blends-common libkdf5-2 libncbi-vdb2 libncbi-wvdb2 med-config menu\n",
            "Suggested packages:\n",
            "  blends-doc menu-l10n gksu | kde-runtime | ktsuss\n",
            "The following NEW packages will be installed:\n",
            "  blends-common libkdf5-2 libncbi-vdb2 libncbi-wvdb2 med-config menu sra-toolkit\n",
            "0 upgraded, 7 newly installed, 0 to remove and 45 not upgraded.\n",
            "Need to get 8,290 kB of archives.\n",
            "After this operation, 23.0 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 menu amd64 2.1.47ubuntu4 [354 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 blends-common all 0.7.4ubuntu1 [15.9 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libkdf5-2 amd64 2.11.2+dfsg-4build2 [14.7 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libncbi-vdb2 amd64 2.11.2+dfsg-4build2 [1,364 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libncbi-wvdb2 amd64 2.11.2+dfsg-4build2 [1,252 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy/universe amd64 sra-toolkit amd64 2.11.3+dfsg-1ubuntu1 [5,277 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy/universe amd64 med-config all 3.7.1 [11.6 kB]\n",
            "Fetched 8,290 kB in 3s (2,767 kB/s)\n",
            "Preconfiguring packages ...\n",
            "Selecting previously unselected package menu.\n",
            "(Reading database ... 121753 files and directories currently installed.)\n",
            "Preparing to unpack .../0-menu_2.1.47ubuntu4_amd64.deb ...\n",
            "Unpacking menu (2.1.47ubuntu4) ...\n",
            "Selecting previously unselected package blends-common.\n",
            "Preparing to unpack .../1-blends-common_0.7.4ubuntu1_all.deb ...\n",
            "Unpacking blends-common (0.7.4ubuntu1) ...\n",
            "Selecting previously unselected package libkdf5-2.\n",
            "Preparing to unpack .../2-libkdf5-2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libkdf5-2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package libncbi-vdb2.\n",
            "Preparing to unpack .../3-libncbi-vdb2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libncbi-vdb2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package libncbi-wvdb2.\n",
            "Preparing to unpack .../4-libncbi-wvdb2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libncbi-wvdb2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package sra-toolkit.\n",
            "Preparing to unpack .../5-sra-toolkit_2.11.3+dfsg-1ubuntu1_amd64.deb ...\n",
            "Unpacking sra-toolkit (2.11.3+dfsg-1ubuntu1) ...\n",
            "Selecting previously unselected package med-config.\n",
            "Preparing to unpack .../6-med-config_3.7.1_all.deb ...\n",
            "Unpacking med-config (3.7.1) ...\n",
            "Setting up libncbi-vdb2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up libkdf5-2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up libncbi-wvdb2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up menu (2.1.47ubuntu4) ...\n",
            "Setting up blends-common (0.7.4ubuntu1) ...\n",
            "Setting up sra-toolkit (2.11.3+dfsg-1ubuntu1) ...\n",
            "Setting up med-config (3.7.1) ...\n",
            "Adding group `med' (GID 107) ...\n",
            "Done.\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.4) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "Processing triggers for menu (2.1.47ubuntu4) ...\n",
            "\n",
            "2024-04-01T04:27:51 prefetch.2.11.3: Current preference is set to retrieve SRA Normalized Format files with full base quality scores.\n",
            "2024-04-01T04:27:52 prefetch.2.11.3: 1) Downloading 'SRR1021789'...\n",
            "2024-04-01T04:27:52 prefetch.2.11.3: SRA Normalized Format file is being retrieved, if this is different from your preference, it may be due to current file availability.\n",
            "2024-04-01T04:27:52 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:28:23 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:28:26 prefetch.2.11.3:  'SRR1021789' is valid\n",
            "2024-04-01T04:28:26 prefetch.2.11.3: 1) 'SRR1021789' was downloaded successfully\n",
            "2024-04-01T04:28:46 prefetch.2.11.3: 'SRR1021789' has 25 unresolved dependencies\n",
            "2024-04-01T04:28:46 prefetch.2.11.3: 2) Downloading 'ncbi-acc:NC_000001.9?vdb-ctx=refseq'...\n",
            "2024-04-01T04:28:46 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:28:52 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:28:52 prefetch.2.11.3: 2) 'ncbi-acc:NC_000001.9?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:28:52 prefetch.2.11.3: 3) Downloading 'ncbi-acc:NC_000002.10?vdb-ctx=refseq'...\n",
            "2024-04-01T04:28:52 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:28:57 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:28:57 prefetch.2.11.3: 3) 'ncbi-acc:NC_000002.10?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:28:57 prefetch.2.11.3: 4) Downloading 'ncbi-acc:NC_000003.10?vdb-ctx=refseq'...\n",
            "2024-04-01T04:28:57 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:29:02 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:29:02 prefetch.2.11.3: 4) 'ncbi-acc:NC_000003.10?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:29:02 prefetch.2.11.3: 5) Downloading 'ncbi-acc:NC_000004.10?vdb-ctx=refseq'...\n",
            "2024-04-01T04:29:02 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:29:07 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:29:07 prefetch.2.11.3: 5) 'ncbi-acc:NC_000004.10?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:29:07 prefetch.2.11.3: 6) Downloading 'ncbi-acc:NC_000005.8?vdb-ctx=refseq'...\n",
            "2024-04-01T04:29:07 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:29:12 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:29:12 prefetch.2.11.3: 6) 'ncbi-acc:NC_000005.8?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:29:12 prefetch.2.11.3: 7) Downloading 'ncbi-acc:NC_000006.10?vdb-ctx=refseq'...\n",
            "2024-04-01T04:29:12 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:29:17 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:29:17 prefetch.2.11.3: 7) 'ncbi-acc:NC_000006.10?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:29:17 prefetch.2.11.3: 8) Downloading 'ncbi-acc:NC_000007.12?vdb-ctx=refseq'...\n",
            "2024-04-01T04:29:17 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:29:22 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:29:22 prefetch.2.11.3: 8) 'ncbi-acc:NC_000007.12?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:29:22 prefetch.2.11.3: 9) Downloading 'ncbi-acc:NC_000008.9?vdb-ctx=refseq'...\n",
            "2024-04-01T04:29:22 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:29:26 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:29:26 prefetch.2.11.3: 9) 'ncbi-acc:NC_000008.9?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:29:26 prefetch.2.11.3: 10) Downloading 'ncbi-acc:NC_000009.10?vdb-ctx=refseq'...\n",
            "2024-04-01T04:29:26 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:29:30 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:29:30 prefetch.2.11.3: 10) 'ncbi-acc:NC_000009.10?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:29:30 prefetch.2.11.3: 11) Downloading 'ncbi-acc:NC_000010.9?vdb-ctx=refseq'...\n",
            "2024-04-01T04:29:30 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:29:34 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:29:34 prefetch.2.11.3: 11) 'ncbi-acc:NC_000010.9?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:29:34 prefetch.2.11.3: 12) Downloading 'ncbi-acc:NC_000011.8?vdb-ctx=refseq'...\n",
            "2024-04-01T04:29:34 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:29:38 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:29:38 prefetch.2.11.3: 12) 'ncbi-acc:NC_000011.8?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:29:38 prefetch.2.11.3: 13) Downloading 'ncbi-acc:NC_000012.10?vdb-ctx=refseq'...\n",
            "2024-04-01T04:29:38 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:29:42 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:29:42 prefetch.2.11.3: 13) 'ncbi-acc:NC_000012.10?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:29:42 prefetch.2.11.3: 14) Downloading 'ncbi-acc:NC_000013.9?vdb-ctx=refseq'...\n",
            "2024-04-01T04:29:42 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:29:45 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:29:45 prefetch.2.11.3: 14) 'ncbi-acc:NC_000013.9?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:29:45 prefetch.2.11.3: 15) Downloading 'ncbi-acc:NC_000014.7?vdb-ctx=refseq'...\n",
            "2024-04-01T04:29:45 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:29:48 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:29:48 prefetch.2.11.3: 15) 'ncbi-acc:NC_000014.7?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:29:48 prefetch.2.11.3: 16) Downloading 'ncbi-acc:NC_000015.8?vdb-ctx=refseq'...\n",
            "2024-04-01T04:29:48 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:29:52 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:29:52 prefetch.2.11.3: 16) 'ncbi-acc:NC_000015.8?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:29:52 prefetch.2.11.3: 17) Downloading 'ncbi-acc:NC_000016.8?vdb-ctx=refseq'...\n",
            "2024-04-01T04:29:52 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:29:55 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:29:55 prefetch.2.11.3: 17) 'ncbi-acc:NC_000016.8?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:29:55 prefetch.2.11.3: 18) Downloading 'ncbi-acc:NC_000017.9?vdb-ctx=refseq'...\n",
            "2024-04-01T04:29:55 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:29:58 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:29:58 prefetch.2.11.3: 18) 'ncbi-acc:NC_000017.9?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:29:58 prefetch.2.11.3: 19) Downloading 'ncbi-acc:NC_000018.8?vdb-ctx=refseq'...\n",
            "2024-04-01T04:29:58 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:30:01 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:30:01 prefetch.2.11.3: 19) 'ncbi-acc:NC_000018.8?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:30:01 prefetch.2.11.3: 20) Downloading 'ncbi-acc:NC_000019.8?vdb-ctx=refseq'...\n",
            "2024-04-01T04:30:01 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:30:04 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:30:04 prefetch.2.11.3: 20) 'ncbi-acc:NC_000019.8?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:30:04 prefetch.2.11.3: 21) Downloading 'ncbi-acc:NC_000020.9?vdb-ctx=refseq'...\n",
            "2024-04-01T04:30:04 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:30:07 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:30:07 prefetch.2.11.3: 21) 'ncbi-acc:NC_000020.9?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:30:07 prefetch.2.11.3: 22) Downloading 'ncbi-acc:NC_000021.7?vdb-ctx=refseq'...\n",
            "2024-04-01T04:30:07 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:30:09 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:30:09 prefetch.2.11.3: 22) 'ncbi-acc:NC_000021.7?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:30:09 prefetch.2.11.3: 23) Downloading 'ncbi-acc:NC_000022.9?vdb-ctx=refseq'...\n",
            "2024-04-01T04:30:09 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:30:12 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:30:12 prefetch.2.11.3: 23) 'ncbi-acc:NC_000022.9?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:30:12 prefetch.2.11.3: 24) Downloading 'ncbi-acc:NC_000023.9?vdb-ctx=refseq'...\n",
            "2024-04-01T04:30:12 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:30:16 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:30:16 prefetch.2.11.3: 24) 'ncbi-acc:NC_000023.9?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:30:16 prefetch.2.11.3: 25) Downloading 'ncbi-acc:NC_000024.8?vdb-ctx=refseq'...\n",
            "2024-04-01T04:30:16 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:30:19 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:30:19 prefetch.2.11.3: 25) 'ncbi-acc:NC_000024.8?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:30:19 prefetch.2.11.3: 26) Downloading 'ncbi-acc:NC_001807.4?vdb-ctx=refseq'...\n",
            "2024-04-01T04:30:19 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:30:20 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:30:20 prefetch.2.11.3: 26) 'ncbi-acc:NC_001807.4?vdb-ctx=refseq' was downloaded successfully\n",
            "\n",
            "2024-04-01T04:30:21 prefetch.2.11.3: Current preference is set to retrieve SRA Normalized Format files with full base quality scores.\n",
            "2024-04-01T04:30:21 prefetch.2.11.3: 1) Downloading 'SRR1021804'...\n",
            "2024-04-01T04:30:21 prefetch.2.11.3: SRA Normalized Format file is being retrieved, if this is different from your preference, it may be due to current file availability.\n",
            "2024-04-01T04:30:21 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:30:44 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:30:46 prefetch.2.11.3:  'SRR1021804' is valid\n",
            "2024-04-01T04:30:46 prefetch.2.11.3: 1) 'SRR1021804' was downloaded successfully\n",
            "2024-04-01T04:31:06 prefetch.2.11.3: 'SRR1021804' has 25 unresolved dependencies\n",
            "2024-04-01T04:31:06 prefetch.2.11.3: 2) Downloading 'ncbi-acc:NC_000001.9?vdb-ctx=refseq'...\n",
            "2024-04-01T04:31:06 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:31:12 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:31:12 prefetch.2.11.3: 2) 'ncbi-acc:NC_000001.9?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:31:12 prefetch.2.11.3: 3) Downloading 'ncbi-acc:NC_000002.10?vdb-ctx=refseq'...\n",
            "2024-04-01T04:31:12 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:31:18 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:31:18 prefetch.2.11.3: 3) 'ncbi-acc:NC_000002.10?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:31:18 prefetch.2.11.3: 4) Downloading 'ncbi-acc:NC_000003.10?vdb-ctx=refseq'...\n",
            "2024-04-01T04:31:18 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:31:24 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:31:24 prefetch.2.11.3: 4) 'ncbi-acc:NC_000003.10?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:31:24 prefetch.2.11.3: 5) Downloading 'ncbi-acc:NC_000004.10?vdb-ctx=refseq'...\n",
            "2024-04-01T04:31:24 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:31:29 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:31:29 prefetch.2.11.3: 5) 'ncbi-acc:NC_000004.10?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:31:29 prefetch.2.11.3: 6) Downloading 'ncbi-acc:NC_000005.8?vdb-ctx=refseq'...\n",
            "2024-04-01T04:31:29 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:31:34 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:31:34 prefetch.2.11.3: 6) 'ncbi-acc:NC_000005.8?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:31:34 prefetch.2.11.3: 7) Downloading 'ncbi-acc:NC_000006.10?vdb-ctx=refseq'...\n",
            "2024-04-01T04:31:34 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:31:39 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:31:39 prefetch.2.11.3: 7) 'ncbi-acc:NC_000006.10?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:31:39 prefetch.2.11.3: 8) Downloading 'ncbi-acc:NC_000007.12?vdb-ctx=refseq'...\n",
            "2024-04-01T04:31:39 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:31:44 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:31:44 prefetch.2.11.3: 8) 'ncbi-acc:NC_000007.12?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:31:44 prefetch.2.11.3: 9) Downloading 'ncbi-acc:NC_000008.9?vdb-ctx=refseq'...\n",
            "2024-04-01T04:31:44 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:31:48 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:31:48 prefetch.2.11.3: 9) 'ncbi-acc:NC_000008.9?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:31:48 prefetch.2.11.3: 10) Downloading 'ncbi-acc:NC_000009.10?vdb-ctx=refseq'...\n",
            "2024-04-01T04:31:48 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:31:52 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:31:52 prefetch.2.11.3: 10) 'ncbi-acc:NC_000009.10?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:31:52 prefetch.2.11.3: 11) Downloading 'ncbi-acc:NC_000010.9?vdb-ctx=refseq'...\n",
            "2024-04-01T04:31:52 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:31:56 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:31:56 prefetch.2.11.3: 11) 'ncbi-acc:NC_000010.9?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:31:56 prefetch.2.11.3: 12) Downloading 'ncbi-acc:NC_000011.8?vdb-ctx=refseq'...\n",
            "2024-04-01T04:31:56 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:32:00 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:32:00 prefetch.2.11.3: 12) 'ncbi-acc:NC_000011.8?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:32:00 prefetch.2.11.3: 13) Downloading 'ncbi-acc:NC_000012.10?vdb-ctx=refseq'...\n",
            "2024-04-01T04:32:00 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:32:03 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:32:03 prefetch.2.11.3: 13) 'ncbi-acc:NC_000012.10?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:32:03 prefetch.2.11.3: 14) Downloading 'ncbi-acc:NC_000013.9?vdb-ctx=refseq'...\n",
            "2024-04-01T04:32:03 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:32:07 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:32:07 prefetch.2.11.3: 14) 'ncbi-acc:NC_000013.9?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:32:07 prefetch.2.11.3: 15) Downloading 'ncbi-acc:NC_000014.7?vdb-ctx=refseq'...\n",
            "2024-04-01T04:32:07 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:32:10 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:32:10 prefetch.2.11.3: 15) 'ncbi-acc:NC_000014.7?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:32:10 prefetch.2.11.3: 16) Downloading 'ncbi-acc:NC_000015.8?vdb-ctx=refseq'...\n",
            "2024-04-01T04:32:10 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:32:14 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:32:14 prefetch.2.11.3: 16) 'ncbi-acc:NC_000015.8?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:32:14 prefetch.2.11.3: 17) Downloading 'ncbi-acc:NC_000016.8?vdb-ctx=refseq'...\n",
            "2024-04-01T04:32:14 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:32:17 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:32:17 prefetch.2.11.3: 17) 'ncbi-acc:NC_000016.8?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:32:17 prefetch.2.11.3: 18) Downloading 'ncbi-acc:NC_000017.9?vdb-ctx=refseq'...\n",
            "2024-04-01T04:32:17 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:32:21 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:32:21 prefetch.2.11.3: 18) 'ncbi-acc:NC_000017.9?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:32:21 prefetch.2.11.3: 19) Downloading 'ncbi-acc:NC_000018.8?vdb-ctx=refseq'...\n",
            "2024-04-01T04:32:21 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:32:24 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:32:24 prefetch.2.11.3: 19) 'ncbi-acc:NC_000018.8?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:32:24 prefetch.2.11.3: 20) Downloading 'ncbi-acc:NC_000019.8?vdb-ctx=refseq'...\n",
            "2024-04-01T04:32:24 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:32:27 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:32:27 prefetch.2.11.3: 20) 'ncbi-acc:NC_000019.8?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:32:27 prefetch.2.11.3: 21) Downloading 'ncbi-acc:NC_000020.9?vdb-ctx=refseq'...\n",
            "2024-04-01T04:32:27 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:32:30 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:32:30 prefetch.2.11.3: 21) 'ncbi-acc:NC_000020.9?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:32:30 prefetch.2.11.3: 22) Downloading 'ncbi-acc:NC_000021.7?vdb-ctx=refseq'...\n",
            "2024-04-01T04:32:30 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:32:32 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:32:32 prefetch.2.11.3: 22) 'ncbi-acc:NC_000021.7?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:32:32 prefetch.2.11.3: 23) Downloading 'ncbi-acc:NC_000022.9?vdb-ctx=refseq'...\n",
            "2024-04-01T04:32:32 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:32:35 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:32:35 prefetch.2.11.3: 23) 'ncbi-acc:NC_000022.9?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:32:35 prefetch.2.11.3: 24) Downloading 'ncbi-acc:NC_000023.9?vdb-ctx=refseq'...\n",
            "2024-04-01T04:32:35 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:32:39 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:32:39 prefetch.2.11.3: 24) 'ncbi-acc:NC_000023.9?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:32:39 prefetch.2.11.3: 25) Downloading 'ncbi-acc:NC_000024.8?vdb-ctx=refseq'...\n",
            "2024-04-01T04:32:39 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:32:42 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:32:42 prefetch.2.11.3: 25) 'ncbi-acc:NC_000024.8?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:32:42 prefetch.2.11.3: 26) Downloading 'ncbi-acc:NC_001807.4?vdb-ctx=refseq'...\n",
            "2024-04-01T04:32:42 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2024-04-01T04:32:43 prefetch.2.11.3:  HTTPS download succeed\n",
            "2024-04-01T04:32:43 prefetch.2.11.3: 26) 'ncbi-acc:NC_001807.4?vdb-ctx=refseq' was downloaded successfully\n",
            "2024-04-01T04:32:44 fastq-dump.2.11.3 err: libs/vfs/names4-response.c:1944:ContainerProcessStatus: name not found while resolving query within virtual file system module - failed to resolve accession '/root/ncbi/public/sra/SRR1021789.sra' - Cannot resolve accession ( 404 )\n",
            "2024-04-01T04:32:44 fastq-dump.2.11.3 err: libs/vfs/manager.c:2277:VFSManagerOpenDirectoryReadKfs: item not found while constructing within virtual database module - the path '/root/ncbi/public/sra/SRR1021789.sra' cannot be opened as database or table\n",
            "fastq-dump quit with error code 3\n",
            "2024-04-01T04:32:45 fastq-dump.2.11.3 err: libs/vfs/names4-response.c:1944:ContainerProcessStatus: name not found while resolving query within virtual file system module - failed to resolve accession '/root/ncbi/public/sra/SRR1021804.sra' - Cannot resolve accession ( 404 )\n",
            "2024-04-01T04:32:45 fastq-dump.2.11.3 err: libs/vfs/manager.c:2277:VFSManagerOpenDirectoryReadKfs: item not found while constructing within virtual database module - the path '/root/ncbi/public/sra/SRR1021804.sra' cannot be opened as database or table\n",
            "fastq-dump quit with error code 3\n"
          ]
        }
      ],
      "source": [
        "# Install sra-tools to download and extract FASTQ files from SRA\n",
        "!apt-get install sra-toolkit\n",
        "\n",
        "# Download SRA files using prefetch (part of sra-tools)\n",
        "!prefetch SRR1021789\n",
        "!prefetch SRR1021804\n",
        "\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "# Convert downloaded SRA files to FASTQ format\n",
        "!fastq-dump --split-files /content/SRR1021789/SRR1021789.sra\n",
        "!fastq-dump --split-files /content/SRR1021804/SRR1021804.sra\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "qWEeIsVA6Tv9",
        "outputId": "e59870da-6abb-40f2-b03a-4af8131ce66c"
      },
      "execution_count": 3,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Read 29151735 spots for /content/SRR1021789/SRR1021789.sra\n",
            "Written 29151735 spots for /content/SRR1021789/SRR1021789.sra\n",
            "Read 22196096 spots for /content/SRR1021804/SRR1021804.sra\n",
            "Written 22196096 spots for /content/SRR1021804/SRR1021804.sra\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!apt-get install fastqc\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ChE0MSMa6TxU",
        "outputId": "ee54af54-dfac-474a-a5bd-eb3c0ac48ccc"
      },
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  default-jre default-jre-headless fonts-dejavu-core fonts-dejavu-extra libapache-pom-java\n",
            "  libargs4j-java libatk-wrapper-java libatk-wrapper-java-jni libcommons-compress-java\n",
            "  libcommons-io-java libcommons-jexl2-java libcommons-lang3-java libcommons-logging-java\n",
            "  libcommons-math3-java libcommons-parent-java libfindbin-libs-perl libfontenc1 libhtsjdk-java\n",
            "  libjbzip2-java libjson-simple-java libngs-java libngs-sdk-dev libngs-sdk2 libsis-base-java\n",
            "  libsis-base-jni libsis-jhdf5-java libsis-jhdf5-jni libsnappy-java libsnappy-jni libxkbfile1\n",
            "  libxtst6 libxxf86dga1 openjdk-11-jre x11-utils\n",
            "Suggested packages:\n",
            "  libxz-java libcommons-io-java-doc libcommons-lang3-java-doc libavalon-framework-java\n",
            "  libcommons-logging-java-doc libexcalibur-logkit-java liblog4j1.2-java picard-tools\n",
            "  libjson-simple-doc mesa-utils\n",
            "The following NEW packages will be installed:\n",
            "  default-jre default-jre-headless fastqc fonts-dejavu-core fonts-dejavu-extra libapache-pom-java\n",
            "  libargs4j-java libatk-wrapper-java libatk-wrapper-java-jni libcommons-compress-java\n",
            "  libcommons-io-java libcommons-jexl2-java libcommons-lang3-java libcommons-logging-java\n",
            "  libcommons-math3-java libcommons-parent-java libfindbin-libs-perl libfontenc1 libhtsjdk-java\n",
            "  libjbzip2-java libjson-simple-java libngs-java libngs-sdk-dev libngs-sdk2 libsis-base-java\n",
            "  libsis-base-jni libsis-jhdf5-java libsis-jhdf5-jni libsnappy-java libsnappy-jni libxkbfile1\n",
            "  libxtst6 libxxf86dga1 openjdk-11-jre x11-utils\n",
            "0 upgraded, 35 newly installed, 0 to remove and 45 not upgraded.\n",
            "Need to get 11.8 MB of archives.\n",
            "After this operation, 24.8 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/main amd64 default-jre-headless amd64 2:1.11-72build2 [3,042 B]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxtst6 amd64 2:1.2.3-1build4 [13.4 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 openjdk-11-jre amd64 11.0.22+7-0ubuntu2~22.04.1 [214 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy/main amd64 default-jre amd64 2:1.11-72build2 [896 B]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-math3-java all 3.6.1-3 [2,016 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libjbzip2-java all 0.9.1-7 [42.8 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libapache-pom-java all 18-1 [4,720 B]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-parent-java all 43-1 [10.8 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-compress-java all 1.21-1 [575 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-logging-java all 1.2-2 [60.3 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-jexl2-java all 2.1.1-5 [251 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libjson-simple-java all 2.3.0-1 [61.0 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libngs-sdk2 amd64 2.11.2-3 [65.6 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libngs-sdk-dev amd64 2.11.2-3 [111 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libngs-java amd64 2.11.2-3 [75.8 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsnappy-jni amd64 1.1.8.3-1 [6,756 B]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsnappy-java all 1.1.8.3-1 [93.9 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhtsjdk-java all 2.24.1+dfsg-1 [1,644 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libfindbin-libs-perl all 2.190.02-1 [23.2 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libargs4j-java all 2.33-1.1 [137 kB]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-io-java all 2.11.0-2 [297 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-lang3-java all 3.11-1 [526 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-base-jni amd64 18.09~pre1+git20180928.45fbd31+dfsg-3 [12.7 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-base-java all 18.09~pre1+git20180928.45fbd31+dfsg-3 [96.4 kB]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-jhdf5-jni amd64 19.04.0+dfsg-4 [96.6 kB]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-jhdf5-java all 19.04.0+dfsg-4 [1,490 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu jammy/universe amd64 fastqc all 0.11.9+dfsg-5 [370 kB]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-core all 2.37-2build1 [1,041 kB]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-extra all 2.37-2build1 [2,041 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu jammy/main amd64 libfontenc1 amd64 1:1.1.4-1build3 [14.7 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxkbfile1 amd64 1:1.1.0-1build3 [71.8 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxxf86dga1 amd64 2:1.1.5-0ubuntu3 [12.6 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu jammy/main amd64 x11-utils amd64 7.7+5build2 [206 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java all 0.38.0-5build1 [53.1 kB]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java-jni amd64 0.38.0-5build1 [49.0 kB]\n",
            "Fetched 11.8 MB in 8s (1,410 kB/s)\n",
            "Extracting templates from packages: 100%\n",
            "Selecting previously unselected package default-jre-headless.\n",
            "(Reading database ... 122103 files and directories currently installed.)\n",
            "Preparing to unpack .../00-default-jre-headless_2%3a1.11-72build2_amd64.deb ...\n",
            "Unpacking default-jre-headless (2:1.11-72build2) ...\n",
            "Selecting previously unselected package libxtst6:amd64.\n",
            "Preparing to unpack .../01-libxtst6_2%3a1.2.3-1build4_amd64.deb ...\n",
            "Unpacking libxtst6:amd64 (2:1.2.3-1build4) ...\n",
            "Selecting previously unselected package openjdk-11-jre:amd64.\n",
            "Preparing to unpack .../02-openjdk-11-jre_11.0.22+7-0ubuntu2~22.04.1_amd64.deb ...\n",
            "Unpacking openjdk-11-jre:amd64 (11.0.22+7-0ubuntu2~22.04.1) ...\n",
            "Selecting previously unselected package default-jre.\n",
            "Preparing to unpack .../03-default-jre_2%3a1.11-72build2_amd64.deb ...\n",
            "Unpacking default-jre (2:1.11-72build2) ...\n",
            "Selecting previously unselected package libcommons-math3-java.\n",
            "Preparing to unpack .../04-libcommons-math3-java_3.6.1-3_all.deb ...\n",
            "Unpacking libcommons-math3-java (3.6.1-3) ...\n",
            "Selecting previously unselected package libjbzip2-java.\n",
            "Preparing to unpack .../05-libjbzip2-java_0.9.1-7_all.deb ...\n",
            "Unpacking libjbzip2-java (0.9.1-7) ...\n",
            "Selecting previously unselected package libapache-pom-java.\n",
            "Preparing to unpack .../06-libapache-pom-java_18-1_all.deb ...\n",
            "Unpacking libapache-pom-java (18-1) ...\n",
            "Selecting previously unselected package libcommons-parent-java.\n",
            "Preparing to unpack .../07-libcommons-parent-java_43-1_all.deb ...\n",
            "Unpacking libcommons-parent-java (43-1) ...\n",
            "Selecting previously unselected package libcommons-compress-java.\n",
            "Preparing to unpack .../08-libcommons-compress-java_1.21-1_all.deb ...\n",
            "Unpacking libcommons-compress-java (1.21-1) ...\n",
            "Selecting previously unselected package libcommons-logging-java.\n",
            "Preparing to unpack .../09-libcommons-logging-java_1.2-2_all.deb ...\n",
            "Unpacking libcommons-logging-java (1.2-2) ...\n",
            "Selecting previously unselected package libcommons-jexl2-java.\n",
            "Preparing to unpack .../10-libcommons-jexl2-java_2.1.1-5_all.deb ...\n",
            "Unpacking libcommons-jexl2-java (2.1.1-5) ...\n",
            "Selecting previously unselected package libjson-simple-java.\n",
            "Preparing to unpack .../11-libjson-simple-java_2.3.0-1_all.deb ...\n",
            "Unpacking libjson-simple-java (2.3.0-1) ...\n",
            "Selecting previously unselected package libngs-sdk2:amd64.\n",
            "Preparing to unpack .../12-libngs-sdk2_2.11.2-3_amd64.deb ...\n",
            "Unpacking libngs-sdk2:amd64 (2.11.2-3) ...\n",
            "Selecting previously unselected package libngs-sdk-dev:amd64.\n",
            "Preparing to unpack .../13-libngs-sdk-dev_2.11.2-3_amd64.deb ...\n",
            "Unpacking libngs-sdk-dev:amd64 (2.11.2-3) ...\n",
            "Selecting previously unselected package libngs-java:amd64.\n",
            "Preparing to unpack .../14-libngs-java_2.11.2-3_amd64.deb ...\n",
            "Unpacking libngs-java:amd64 (2.11.2-3) ...\n",
            "Selecting previously unselected package libsnappy-jni.\n",
            "Preparing to unpack .../15-libsnappy-jni_1.1.8.3-1_amd64.deb ...\n",
            "Unpacking libsnappy-jni (1.1.8.3-1) ...\n",
            "Selecting previously unselected package libsnappy-java.\n",
            "Preparing to unpack .../16-libsnappy-java_1.1.8.3-1_all.deb ...\n",
            "Unpacking libsnappy-java (1.1.8.3-1) ...\n",
            "Selecting previously unselected package libhtsjdk-java.\n",
            "Preparing to unpack .../17-libhtsjdk-java_2.24.1+dfsg-1_all.deb ...\n",
            "Unpacking libhtsjdk-java (2.24.1+dfsg-1) ...\n",
            "Selecting previously unselected package libfindbin-libs-perl.\n",
            "Preparing to unpack .../18-libfindbin-libs-perl_2.190.02-1_all.deb ...\n",
            "Unpacking libfindbin-libs-perl (2.190.02-1) ...\n",
            "Selecting previously unselected package libargs4j-java.\n",
            "Preparing to unpack .../19-libargs4j-java_2.33-1.1_all.deb ...\n",
            "Unpacking libargs4j-java (2.33-1.1) ...\n",
            "Selecting previously unselected package libcommons-io-java.\n",
            "Preparing to unpack .../20-libcommons-io-java_2.11.0-2_all.deb ...\n",
            "Unpacking libcommons-io-java (2.11.0-2) ...\n",
            "Selecting previously unselected package libcommons-lang3-java.\n",
            "Preparing to unpack .../21-libcommons-lang3-java_3.11-1_all.deb ...\n",
            "Unpacking libcommons-lang3-java (3.11-1) ...\n",
            "Selecting previously unselected package libsis-base-jni.\n",
            "Preparing to unpack .../22-libsis-base-jni_18.09~pre1+git20180928.45fbd31+dfsg-3_amd64.deb ...\n",
            "Unpacking libsis-base-jni (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Selecting previously unselected package libsis-base-java.\n",
            "Preparing to unpack .../23-libsis-base-java_18.09~pre1+git20180928.45fbd31+dfsg-3_all.deb ...\n",
            "Unpacking libsis-base-java (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Selecting previously unselected package libsis-jhdf5-jni.\n",
            "Preparing to unpack .../24-libsis-jhdf5-jni_19.04.0+dfsg-4_amd64.deb ...\n",
            "Unpacking libsis-jhdf5-jni (19.04.0+dfsg-4) ...\n",
            "Selecting previously unselected package libsis-jhdf5-java.\n",
            "Preparing to unpack .../25-libsis-jhdf5-java_19.04.0+dfsg-4_all.deb ...\n",
            "Unpacking libsis-jhdf5-java (19.04.0+dfsg-4) ...\n",
            "Selecting previously unselected package fastqc.\n",
            "Preparing to unpack .../26-fastqc_0.11.9+dfsg-5_all.deb ...\n",
            "Unpacking fastqc (0.11.9+dfsg-5) ...\n",
            "Selecting previously unselected package fonts-dejavu-core.\n",
            "Preparing to unpack .../27-fonts-dejavu-core_2.37-2build1_all.deb ...\n",
            "Unpacking fonts-dejavu-core (2.37-2build1) ...\n",
            "Selecting previously unselected package fonts-dejavu-extra.\n",
            "Preparing to unpack .../28-fonts-dejavu-extra_2.37-2build1_all.deb ...\n",
            "Unpacking fonts-dejavu-extra (2.37-2build1) ...\n",
            "Selecting previously unselected package libfontenc1:amd64.\n",
            "Preparing to unpack .../29-libfontenc1_1%3a1.1.4-1build3_amd64.deb ...\n",
            "Unpacking libfontenc1:amd64 (1:1.1.4-1build3) ...\n",
            "Selecting previously unselected package libxkbfile1:amd64.\n",
            "Preparing to unpack .../30-libxkbfile1_1%3a1.1.0-1build3_amd64.deb ...\n",
            "Unpacking libxkbfile1:amd64 (1:1.1.0-1build3) ...\n",
            "Selecting previously unselected package libxxf86dga1:amd64.\n",
            "Preparing to unpack .../31-libxxf86dga1_2%3a1.1.5-0ubuntu3_amd64.deb ...\n",
            "Unpacking libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n",
            "Selecting previously unselected package x11-utils.\n",
            "Preparing to unpack .../32-x11-utils_7.7+5build2_amd64.deb ...\n",
            "Unpacking x11-utils (7.7+5build2) ...\n",
            "Selecting previously unselected package libatk-wrapper-java.\n",
            "Preparing to unpack .../33-libatk-wrapper-java_0.38.0-5build1_all.deb ...\n",
            "Unpacking libatk-wrapper-java (0.38.0-5build1) ...\n",
            "Selecting previously unselected package libatk-wrapper-java-jni:amd64.\n",
            "Preparing to unpack .../34-libatk-wrapper-java-jni_0.38.0-5build1_amd64.deb ...\n",
            "Unpacking libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n",
            "Setting up default-jre-headless (2:1.11-72build2) ...\n",
            "Setting up libxtst6:amd64 (2:1.2.3-1build4) ...\n",
            "Setting up libngs-sdk2:amd64 (2.11.2-3) ...\n",
            "Setting up libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n",
            "Setting up openjdk-11-jre:amd64 (11.0.22+7-0ubuntu2~22.04.1) ...\n",
            "Setting up libfindbin-libs-perl (2.190.02-1) ...\n",
            "Setting up default-jre (2:1.11-72build2) ...\n",
            "Setting up libjson-simple-java (2.3.0-1) ...\n",
            "Setting up libsis-base-jni (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Setting up libfontenc1:amd64 (1:1.1.4-1build3) ...\n",
            "Setting up libapache-pom-java (18-1) ...\n",
            "Setting up fonts-dejavu-core (2.37-2build1) ...\n",
            "Setting up libjbzip2-java (0.9.1-7) ...\n",
            "Setting up libngs-sdk-dev:amd64 (2.11.2-3) ...\n",
            "Setting up fonts-dejavu-extra (2.37-2build1) ...\n",
            "Setting up libcommons-math3-java (3.6.1-3) ...\n",
            "Setting up libsis-jhdf5-jni (19.04.0+dfsg-4) ...\n",
            "Setting up libxkbfile1:amd64 (1:1.1.0-1build3) ...\n",
            "Setting up libngs-java:amd64 (2.11.2-3) ...\n",
            "Setting up libsnappy-jni (1.1.8.3-1) ...\n",
            "Setting up libargs4j-java (2.33-1.1) ...\n",
            "Setting up libcommons-parent-java (43-1) ...\n",
            "Setting up libcommons-logging-java (1.2-2) ...\n",
            "Setting up libcommons-jexl2-java (2.1.1-5) ...\n",
            "Setting up libcommons-lang3-java (3.11-1) ...\n",
            "Setting up x11-utils (7.7+5build2) ...\n",
            "Setting up libsnappy-java (1.1.8.3-1) ...\n",
            "Setting up libatk-wrapper-java (0.38.0-5build1) ...\n",
            "Setting up libcommons-compress-java (1.21-1) ...\n",
            "Setting up libcommons-io-java (2.11.0-2) ...\n",
            "Setting up libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n",
            "Setting up libsis-base-java (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Setting up libhtsjdk-java (2.24.1+dfsg-1) ...\n",
            "Setting up libsis-jhdf5-java (19.04.0+dfsg-4) ...\n",
            "Setting up fastqc (0.11.9+dfsg-5) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for menu (2.1.47ubuntu4) ...\n",
            "Processing triggers for fontconfig (2.13.1-4.2ubuntu5) ...\n",
            "Processing triggers for hicolor-icon-theme (0.17-2) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.4) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!fastqc /content/SRR1021789_1.fastq\n",
        "!fastqc /content/SRR1021804_1.fastq\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0XGlbVns6Ty1",
        "outputId": "220f6e0c-b796-4efe-a6df-26af18833118"
      },
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of SRR1021789_1.fastq\n",
            "Approx 5% complete for SRR1021789_1.fastq\n",
            "Approx 10% complete for SRR1021789_1.fastq\n",
            "Approx 15% complete for SRR1021789_1.fastq\n",
            "Approx 20% complete for SRR1021789_1.fastq\n",
            "Approx 25% complete for SRR1021789_1.fastq\n",
            "Approx 30% complete for SRR1021789_1.fastq\n",
            "Approx 35% complete for SRR1021789_1.fastq\n",
            "Approx 40% complete for SRR1021789_1.fastq\n",
            "Approx 45% complete for SRR1021789_1.fastq\n",
            "Approx 50% complete for SRR1021789_1.fastq\n",
            "Approx 55% complete for SRR1021789_1.fastq\n",
            "Approx 60% complete for SRR1021789_1.fastq\n",
            "Approx 65% complete for SRR1021789_1.fastq\n",
            "Approx 70% complete for SRR1021789_1.fastq\n",
            "Approx 75% complete for SRR1021789_1.fastq\n",
            "Approx 80% complete for SRR1021789_1.fastq\n",
            "Approx 85% complete for SRR1021789_1.fastq\n",
            "Approx 90% complete for SRR1021789_1.fastq\n",
            "Approx 95% complete for SRR1021789_1.fastq\n",
            "Analysis complete for SRR1021789_1.fastq\n",
            "Started analysis of SRR1021804_1.fastq\n",
            "Approx 5% complete for SRR1021804_1.fastq\n",
            "Approx 10% complete for SRR1021804_1.fastq\n",
            "Approx 15% complete for SRR1021804_1.fastq\n",
            "Approx 20% complete for SRR1021804_1.fastq\n",
            "Approx 25% complete for SRR1021804_1.fastq\n",
            "Approx 30% complete for SRR1021804_1.fastq\n",
            "Approx 35% complete for SRR1021804_1.fastq\n",
            "Approx 40% complete for SRR1021804_1.fastq\n",
            "Approx 45% complete for SRR1021804_1.fastq\n",
            "Approx 50% complete for SRR1021804_1.fastq\n",
            "Approx 55% complete for SRR1021804_1.fastq\n",
            "Approx 60% complete for SRR1021804_1.fastq\n",
            "Approx 65% complete for SRR1021804_1.fastq\n",
            "Approx 70% complete for SRR1021804_1.fastq\n",
            "Approx 75% complete for SRR1021804_1.fastq\n",
            "Approx 80% complete for SRR1021804_1.fastq\n",
            "Approx 85% complete for SRR1021804_1.fastq\n",
            "Approx 90% complete for SRR1021804_1.fastq\n",
            "Approx 95% complete for SRR1021804_1.fastq\n",
            "Analysis complete for SRR1021804_1.fastq\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!apt-get install bowtie2\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "vEPhtYFH6T0i",
        "outputId": "bf6d4b62-5c84-436e-f690-7661df544624"
      },
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "Suggested packages:\n",
            "  bowtie2-examples\n",
            "The following NEW packages will be installed:\n",
            "  bowtie2\n",
            "0 upgraded, 1 newly installed, 0 to remove and 45 not upgraded.\n",
            "Need to get 1,169 kB of archives.\n",
            "After this operation, 3,735 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 bowtie2 amd64 2.4.4-1 [1,169 kB]\n",
            "Fetched 1,169 kB in 2s (579 kB/s)\n",
            "Selecting previously unselected package bowtie2.\n",
            "(Reading database ... 122702 files and directories currently installed.)\n",
            "Preparing to unpack .../bowtie2_2.4.4-1_amd64.deb ...\n",
            "Unpacking bowtie2 (2.4.4-1) ...\n",
            "Setting up bowtie2 (2.4.4-1) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Hz6opiD3BY17",
        "outputId": "8e7486e0-e4a6-4718-dde1-421116c6382c"
      },
      "execution_count": 21,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Mounted at /content/drive\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!unzip /content/drive/MyDrive/hg19.zip -d /content/drive/MyDrive/hg19_index\n",
        "\n"
      ],
      "metadata": {
        "id": "8CzbmK8qSR1C"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!bowtie2 -p 2 -x /content/drive/MyDrive/hg19_index/hg19 -U /content/SRR1021789_1.fastq -S /content/SRR1021789_aligned.sam\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "nlPOPECCSR3H",
        "outputId": "e421eab1-fd98-4743-8e8c-49dd33bd5359"
      },
      "execution_count": 24,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "29151735 reads; of these:\n",
            "  29151735 (100.00%) were unpaired; of these:\n",
            "    95760 (0.33%) aligned 0 times\n",
            "    26027413 (89.28%) aligned exactly 1 time\n",
            "    3028562 (10.39%) aligned >1 times\n",
            "99.67% overall alignment rate\n",
            "22196096 reads; of these:\n",
            "  22196096 (100.00%) were unpaired; of these:\n",
            "    67084 (0.30%) aligned 0 times\n",
            "    19174120 (86.39%) aligned exactly 1 time\n",
            "    2954892 (13.31%) aligned >1 times\n",
            "99.70% overall alignment rate\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!bowtie2 -p 2 -x /content/drive/MyDrive/hg19_index/hg19 -U /content/SRR1021804_1.fastq -S /content/SRR1021804_aligned.sam\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ywVdUf6KlKNi",
        "outputId": "49c3aff5-a5bf-4a04-822b-d8e4fb4fd5aa"
      },
      "execution_count": 28,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "22196096 reads; of these:\n",
            "  22196096 (100.00%) were unpaired; of these:\n",
            "    67084 (0.30%) aligned 0 times\n",
            "    19174120 (86.39%) aligned exactly 1 time\n",
            "    2954892 (13.31%) aligned >1 times\n",
            "99.70% overall alignment rate\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install MACS2\n",
        "\n",
        "!macs2 callpeak -t /content/SRR1021789_aligned.sam -c /content/SRR1021804_aligned.sam -f SAM -g hs -n estrogen\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "URsdnwXTBY5F",
        "outputId": "65fc9de6-1d36-4919-8567-17a3b3910f32"
      },
      "execution_count": 29,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: MACS2 in /usr/local/lib/python3.10/dist-packages (2.2.9.1)\n",
            "Requirement already satisfied: numpy>=1.19 in /usr/local/lib/python3.10/dist-packages (from MACS2) (1.25.2)\n",
            "Requirement already satisfied: Cython~=0.29 in /usr/local/lib/python3.10/dist-packages (from MACS2) (0.29.37)\n",
            "INFO  @ Mon, 01 Apr 2024 07:51:07: \n",
            "# Command line: callpeak -t /content/SRR1021789_aligned.sam -c /content/SRR1021804_aligned.sam -f SAM -g hs -n estrogen\n",
            "# ARGUMENTS LIST:\n",
            "# name = estrogen\n",
            "# format = SAM\n",
            "# ChIP-seq file = ['/content/SRR1021789_aligned.sam']\n",
            "# control file = ['/content/SRR1021804_aligned.sam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff = 5.00e-02\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is off\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:07: #1 read tag files... \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:07: #1 read treatment tags... \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:09:  1000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:11:  2000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:13:  3000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:15:  4000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:17:  5000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:19:  6000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:22:  7000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:24:  8000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:26:  9000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:28:  10000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:30:  11000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:32:  12000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:34:  13000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:36:  14000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:38:  15000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:40:  16000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:42:  17000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:44:  18000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:46:  19000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:48:  20000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:50:  21000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:52:  22000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:54:  23000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:56:  24000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:51:58:  25000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:00:  26000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:02:  27000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:04:  28000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:06:  29000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:07: #1.2 read input tags... \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:09:  1000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:11:  2000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:13:  3000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:15:  4000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:17:  5000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:19:  6000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:21:  7000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:23:  8000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:25:  9000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:27:  10000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:29:  11000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:32:  12000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:34:  13000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:36:  14000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:38:  15000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:40:  16000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:42:  17000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:44:  18000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:46:  19000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:48:  20000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:50:  21000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:52:  22000000 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:53: #1 tag size is determined as 36 bps \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:53: #1 tag size = 36.0 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:53: #1  total tags in treatment: 29055975 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:53: #1 user defined the maximum tags... \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:53: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:53: #1  tags after filtering in treatment: 25295038 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:53: #1  Redundant rate of treatment: 0.13 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:53: #1  total tags in control: 22129012 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:53: #1 user defined the maximum tags... \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:53: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:53: #1  tags after filtering in control: 21655144 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:53: #1  Redundant rate of control: 0.02 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:53: #1 finished! \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:53: #2 Build Peak Model... \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:53: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:58: #2 number of paired peaks: 59583 \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:58: start model_add_line... \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:59: start X-correlation... \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:59: end of X-cor \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:59: #2 finished! \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:59: #2 predicted fragment length is 137 bps \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:59: #2 alternative fragment length(s) may be 137 bps \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:59: #2.2 Generate R script for model : estrogen_model.r \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:59: #3 Call peaks... \n",
            "INFO  @ Mon, 01 Apr 2024 07:52:59: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Mon, 01 Apr 2024 07:54:51: #3 Call peaks for each chromosome... \n",
            "INFO  @ Mon, 01 Apr 2024 07:55:52: #4 Write output xls file... estrogen_peaks.xls \n",
            "INFO  @ Mon, 01 Apr 2024 07:55:52: #4 Write peak in narrowPeak format file... estrogen_peaks.narrowPeak \n",
            "INFO  @ Mon, 01 Apr 2024 07:55:52: #4 Write summits bed file... estrogen_summits.bed \n",
            "INFO  @ Mon, 01 Apr 2024 07:55:52: Done! \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install pybedtools\n",
        "from pybedtools import BedTool\n",
        "\n",
        "# Convert peak file to BedTool object\n",
        "peaks = BedTool('/content/estrogen_peaks.narrowPeak')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "D7YWcCLvBY9M",
        "outputId": "9289e691-d9d9-44f8-c2ba-30be5c89c859"
      },
      "execution_count": 31,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting pybedtools\n",
            "  Downloading pybedtools-0.9.1.tar.gz (12.5 MB)\n",
            "\u001b[2K     \u001b[90m\u001b[0m \u001b[32m12.5/12.5 MB\u001b[0m \u001b[31m75.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.10/dist-packages (from pybedtools) (1.16.0)\n",
            "Collecting pysam (from pybedtools)\n",
            "  Downloading pysam-0.22.0-cp310-cp310-manylinux_2_28_x86_64.whl (21.9 MB)\n",
            "\u001b[2K     \u001b[90m\u001b[0m \u001b[32m21.9/21.9 MB\u001b[0m \u001b[31m25.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from pybedtools) (1.25.2)\n",
            "Building wheels for collected packages: pybedtools\n",
            "  Building wheel for pybedtools (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pybedtools: filename=pybedtools-0.9.1-cp310-cp310-linux_x86_64.whl size=13997646 sha256=d8c9d5eee23d06b80c0ac8ecccc899c851928b540687f66b6fb1f76d254f1e15\n",
            "  Stored in directory: /root/.cache/pip/wheels/f5/79/76/c2cef660013cdf573214696aec4bc7ec352bf46141682230af\n",
            "Successfully built pybedtools\n",
            "Installing collected packages: pysam, pybedtools\n",
            "Successfully installed pybedtools-0.9.1 pysam-0.22.0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "peaks"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "a7yGMxPtBY_L",
        "outputId": "7fd7e50c-2488-4093-cfac-b0446e20ea03"
      },
      "execution_count": 32,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "<BedTool(/content/estrogen_peaks.narrowPeak)>"
            ]
          },
          "metadata": {},
          "execution_count": 32
        }
      ]
    }
  ]
}